backend/cfg/cfg_regalloc_utils.ml

```diff
@@ -456,3 +461,68 @@ let may_use_stack_operands_everywhere :
   may_use_stack_operands_array map instr.arg;
   may_use_stack_operands_array map instr.res;
   All_spilled_registers_rewritten
+
+let insert_spill_block :
+    Cfg_with_layout.t ->
+    Instruction.t list ->
+    after:Cfg.basic_block ->
+    terminator_id:int ->
+    unit =
+ fun cfg_with_layout spills ~after:predecessor_block ~terminator_id ->
+  let cfg = Cfg_with_layout.cfg cfg_with_layout in
+  let successors =
+    Cfg.successor_labels ~normal:true ~exn:false predecessor_block
+  in
+  let successor_label =
+    match Label.Set.to_seq successors |> List.of_seq with
+    | [label] -> label
+    | [] ->
+      Misc.fatal_errorf
+        "Cannot insert a block after block %a: it has no successors" Label.print
+        predecessor_block.start
+    | _ :: _ :: _ ->
+      Misc.fatal_errorf
+        "Cannot insert a block after block %a: it has more than one successor"
+        Label.print predecessor_block.start
+  in
+  let successor_block = Cfg.get_block_exn cfg successor_label in
+  let last_spill = List.nth spills (List.length spills - 1) in
```

Comment:
Do we really need the `last_spill`
thing? Liveness will be recomputed,
and it is not clear fdo/dbg can't
simply be none.

Reply:
(If we keep `last_spill`, I think we
could go through a utility function
rather than use `length` and `nth`.)

Reply:
About fdo/dbg we should probably ask @gretay-js ?

I'm also not sure about what is or isn't necessary here, so I've added the most precise version in my opinion.